apps/web/components/I18nLanguageHandler.tsx

```diff
@@ -1,16 +1,20 @@
+import parser from "accept-language-parser";
 import { useSession } from "next-auth/react";
 import { useTranslation } from "next-i18next";
 import { useEffect } from "react";
 
+import { locales } from "@calcom/lib/i18n";
 import { trpc } from "@calcom/trpc/react";
 
 // eslint-disable-next-line turbo/no-undeclared-env-vars
-const vercelCommitHash = process.env.NEXT_PUBLIC_VERCEL_GIT_COMMIT_SHA || "NA";
+const CalComVersion = process.env.NEXT_PUBLIC_VERCEL_GIT_COMMIT_SHA || "NA";
 
-export function useViewerI18n(locale: string) {
+function useViewerI18n(locale: string) {
   return trpc.viewer.public.i18n.useQuery(
-    { locale, CalComVersion: vercelCommitHash },
+    { locale, CalComVersion },
     {
+      initialData: { locale: null, i18n: {} },
+      refetchOnMount: "always",
```

Comment:
I am not sure why we need this. As it's tied to commit hash, shouldn't we just fetch it once?

Reply:
This is due using initialData so data is never undefined. But using it will prevent the initial fetch being made. Hence the referchObMount